sdks/python/apache_beam/transforms/core.py

```diff
@@ -220,15 +222,29 @@ def __init__(self, generator, *args, **kwargs):
     self._gen_args = args
     self._gen_kwargs = kwargs
 
+  def element_generator(self):
+    while self._current_value is not None:
+      yield self._current_value
+
+    raise StopIteration()
+
   def start_bundle(self, context):
-    self._generator = self._generator_def(*self._gen_args, **self._gen_kwargs)
-    self._generator.send(None)
+    self._generator = self._generator_def(
+        self.element_generator(), *self._gen_args, **self._gen_kwargs)
 
   def process(self, context, *args, **kwargs):
-    return self._generator.send((context, args, kwargs))
+    self._current_value = (context, args, kwargs)
```

Comment:
However, I think it's still better syntax to use a generator for 1:1 processing. And I think the arguments for it being better syntax could be extended to `CombineFn` as well (allowing idiomatic usage of user code within the "create accumulator" function e.g.)